### PR TITLE
[Enhancement] doris to support function SESSION_USER()

### LINKED ISSUE #39574: [Enhancement] doris to support function SESSION_USER()
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Doris to support SESSION_USER()  function.

Return the current user name and host name for the MySQL connection

This function is similar to USER() function

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Vallishp**
## Proposed changes

Issue Number: close #39574

Added support for new scalar function session_user()

when run sqlmap tool on Doris master  it gave  error like  session_user() function not found.   mysql support this function.
https://www.w3schools.com/sql/func_mysql_session_user.asp
 Its similar to Doris user() function.

mysql> select  SESSION_USER();
+--------------------+
| session_user()     |
+--------------------+
| 'root'@'127.0.0.1' |
+--------------------+
1 row in set (0.01 sec)

mysql> select  session_user();
+--------------------+
| session_user()     |
+--------------------+
| 'root'@'127.0.0.1' |
+--------------------+
1 row in set (0.00 sec)

mysql>

### CODE REVIEW

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Review by xinyiZzz — DISMISSED** [review]
LGTM

**Comment by xinyiZzz** [comment]
can improve comment, link to the MySQL function docs.

**Review by morrySnow — DISMISSED** [review]

**Review by 924060929 — DISMISSED** [review]

**Comment by morrySnow** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

**Comment by Vallishp** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/SessionUser.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * ScalarFunction 'SessionUser'.
+ */
+public class SessionUser extends ScalarFunction
+        implements LeafExpression, ExplicitlyCastableSignature, AlwaysNotNullable, Nondeterministic {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/SessionUser.java:36]: Nondeterministic trait be removed in PR #39944. u should override `isDeterministic` and let it return `false`

Comment (Vallishp) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/SessionUser.java]: updated.

**Comment by Vallishp** [comment]
run buildall

**Comment by morrySnow** [comment]
run p0

**Review by Nitin-Kashyap — COMMENTED** [review]
LGTM

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Pull request merged**